src/Microsoft.AspNet.Authorization/Infrastructure/AssertionRequirement.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Authorization.Infrastructure
+{
+    public class AssertionRequirement : AuthorizationHandler<AssertionRequirement>, IAuthorizationRequirement
+    {
+        /// <summary>
+        /// Function that is called to handle this requirement
+        /// </summary>
+        public Func<AuthorizationContext, bool> Handler { get; }
+
+        public AssertionRequirement(Func<AuthorizationContext, bool> handleMe)
+        {
+            Handler = handleMe;
+        }
+
+        protected override void Handle(AuthorizationContext context, AssertionRequirement requirement)
+        {
+            if (Handler(context))
```

Comment:
We don't have any try/catch guards today when running AuthZ handlers everywhere else, they just become uncaught exceptions.  

I don't think this one should be special as a result, I think blowing up is fine, they can always wrap it themselves